### PR TITLE
[ut] Pull common logic out into a common makefile.

### PR DESCRIPTION
**Pull request by DHowett-MSFT**
This is the develop side of the Makefile changes from #1348!

Normal use is documented in Foundation/Makefile.

The aggregate makefile has been updated to truly support multiple tests.

---
This change is [Reviewable](https://reviewable.io/reviews/microsoft/winobjc/1352)

### CODE REVIEW

File: tests/unittests/Foundation/Makefile [hunk 1]
```diff
@@ -2,88 +2,17 @@
...
-CLANG := clang++
-OSX_VERSION_MIN := 10.10
+# Built from every .m and .mm file in this directory ...
+UT_FILES := $(wildcard *.m) $(wildcard *.mm)
 
-SRCDIR := .
-SRCFILES := $(wildcard $(SRCDIR)/*.m*)
-RF_SRCFILES := $(wildcard $(SRCDIR)/ReferenceFoundation/*.mm)
+# ... and in ReferenceFoundation.
+UT_FILES += $(wildcard ReferenceFoundation/*.mm)
```
Comment (aballway) [inline tests/unittests/Foundation/Makefile]: .m here?

File: tests/unittests/UT.common.mk [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+CXXFLAGS := -std=c++14
+CFLAGS := $(foreach i,$(_INC),-I $(i))
+CFLAGS += -DTARGET_OS_MAC
+CFLAGS += -DWINOBJCRT_IMPEXP=
+CFLAGS += -DLOGGING_IMPEXP=
+CFLAGS += -mmacosx-version-min=$(OSX_VERSION_MIN)
+CFLAGS += $(UT_CFLAGS)
+
+_UT_FRAMEWORKS := CoreFoundation Foundation $(UT_FRAMEWORKS)
```
Comment (ms-jihua) [inline tests/unittests/UT.common.mk]: >$(UT_FRAMEWORKS) [](start = 44, length = 16)

this doesn't seem to be declared anywhere, is the expectation that we'll have dirty flags when making this makefile?

Comment (DHowett-MSFT) [inline tests/unittests/UT.common.mk]: Some of these can be empry; UT_FRAMEWORKS is one of them.

If `RESOURCES`, `RESOURCE_DIRS`, `FRAMEWORKS`, `CFLAGS`, or `LDFLAGS` are empty, they're ignored.

**Comment by ms-jihua** [comment]
:shipit:

**Comment by DHowett-MSFT** [comment]
Bypassing CI builds as this change does not touch built collateral.

**Pull request merged**